src/devices/src/virtio/mmio.rs

```diff
@@ -343,542 +343,575 @@ impl BusDevice for MmioTransport {
+        acked_features: u64,
+        avail_features: u64,
+        interrupt_evt: EventFd,
+        interrupt_status: Arc<AtomicUsize>,
+        queue_evts: Vec<EventFd>,
+        queues: Vec<Queue>,
+        device_activated: bool,
+        config_bytes: [u8; 0xeff],
+    }
+
+    impl DummyDevice {
+        fn new() -> Self {
+            DummyDevice {
+                acked_features: 0,
+                avail_features: 0,
+                interrupt_evt: EventFd::new(libc::EFD_NONBLOCK).unwrap(),
+                interrupt_status: Arc::new(AtomicUsize::new(0)),
+                queue_evts: vec![
+                    EventFd::new(libc::EFD_NONBLOCK).unwrap(),
+                    EventFd::new(libc::EFD_NONBLOCK).unwrap(),
+                ],
+                queues: vec![Queue::new(16), Queue::new(32)],
+                device_activated: false,
+                config_bytes: [0; 0xeff],
+            }
+        }
+    }
+
+    impl VirtioDevice for DummyDevice {
+        fn device_type(&self) -> u32 {
+            123
+        }
+
+        fn read_config(&self, offset: u64, data: &mut [u8]) {
+            data.copy_from_slice(&self.config_bytes[offset as usize..]);
+        }
+
+        fn write_config(&mut self, offset: u64, data: &[u8]) {
+            for (i, item) in data.iter().enumerate() {
+                self.config_bytes[offset as usize + i] = *item;
+            }
+        }
+
+        fn avail_features(&self) -> u64 {
+            self.avail_features
+        }
+
+        fn acked_features(&self) -> u64 {
+            self.acked_features
+        }
+
+        fn set_acked_features(&mut self, acked_features: u64) {
+            self.acked_features = acked_features;
+        }
+
+        fn activate(&mut self, _mem: GuestMemory) -> ActivateResult {
+            Ok(())
+        }
+
+        fn get_queues(&mut self) -> &mut Vec<Queue> {
+            &mut self.queues
+        }
+
+        fn get_queue_events(&self) -> result::Result<Vec<EventFd>, std::io::Error> {
+            self.queue_evts.iter().map(|evt| evt.try_clone()).collect()
+        }
+
+        fn get_interrupt(&self) -> result::Result<EventFd, std::io::Error> {
+            self.interrupt_evt.try_clone()
+        }
+
+        fn get_interrupt_status(&self) -> Arc<AtomicUsize> {
+            self.interrupt_status.clone()
+        }
+
+        fn is_activated(&self) -> bool {
+            self.device_activated
+        }
+
+        fn set_device_activated(&mut self, device_activated: bool) {
+            self.device_activated = device_activated;
+        }
+
+        fn set_avail_features(&mut self, value: u64) {
+            self.avail_features = value;
+        }
+    }
+
+    fn set_device_status(d: &mut MmioTransport, status: u32) {
+        let mut buf = vec![0; 4];
+        LittleEndian::write_u32(&mut buf[..], status);
+        d.write(0x70, &buf[..]);
+    }
+
+    #[test]
+    fn test_new() {
+        let m = GuestMemory::new(&[(GuestAddress(0), 0x1000)]).unwrap();
```

Comment:
nit-ish: The test code is pretty hard to read when using one letter defines for variables.
We could use `mem`, `transport`, `dummy_device` instead.

Reply:
We can address it in a future PR.

Reply:
Added descriptive names for the above unit tests variables.